golem-worker-service/src/api/validation.rs

```diff
@@ -0,0 +1,93 @@
+use crate::api::ApiDefinition;
+use crate::api::BindingType;
+
+pub fn validate_api_definition(api: &ApiDefinition) -> Result<(), String> {
+    for route in &api.routes {
+        match &route.binding {
+            BindingType::Default { input_type, output_type, .. } => {
+                validate_wit_type(input_type)?;
+                validate_wit_type(output_type)?;
+            }
+            _ => {}
+        }
+    }
+    Ok(())
+}
+
+fn validate_wit_type(wit_type: &str) -> Result<(), String> {
```

Comment:
I haven't seen where this function is used yet, but if we are ever validating WIT types represented as plain strings, we have taken a wrong turn somewhere!

There are strongly-typed representations of WIT types in the project that can be used in place of strings.

Reply:
You are absolutely right! It's my mistake to do it like this! Thanks for pointing out! Changing this too.